lib/ts/controllers/s-modal.ts

```diff
@@ -153,58 +153,69 @@ namespace Stacks {
         }
 
         /**
-         * Binds tab presses on tabbable items such that tabbing only works within the modal
+         * Gets all elements that could be tabbed to in 
          */
-        private handleFocusableElements() {
-            // get all tabbable items
-            var allTabbables = Array.from(this.modalTarget.querySelectorAll("[href], input, select, textarea, button, [tabindex]"))
+        private getAllTabbables() {
+            return <HTMLElement[]>Array.from(this.modalTarget.querySelectorAll("[href], input, select, textarea, button, [tabindex]"))
                 .filter((el: Element) => el.matches(":not([disabled]):not([tabindex='-1'])"));
+        }
 
-            if (!allTabbables.length) {
-                return;
-            }
-
-            var initialFocus = <HTMLElement>allTabbables[0];
+        private firstVisible(elements: HTMLElement[]) {
+            // https://stackoverflow.com/a/21696585
+            return elements.find(el => el.offsetParent !== null);
```

Comment:
optional: This solution is fine for me, but if we wanted to be more thorough, jQuery's hidden selector is actually _really_ small: https://github.com/jquery/jquery/blob/a684e6ba836f7c553968d7d026ed7941e1a612d8/src/css/hiddenVisibleSelectors.js#L9

Reply:
I'm inclined to leave things alone here.  The edge that jQuery _may_ be avoiding is `position: fixed` elements, which shouldn't exist within or be focusable within our modals.

Reply:
Sounds good to me! Your solution is clean and works _now_, so I'm totally ok with merging as-is. If we run into an edge case in the theoretical future, we can address it at that time.